src/main/java/com/commercetools/sync/services/UnresolvedReferencesService.java

```diff
@@ -12,30 +12,45 @@
    * Given product draft keys, this method fetches the persisted drafts waiting to be resolved. If
    * there is no matching draft, an empty {@link Set} will be returned in the returned future.
    *
-   * @param keys the keys of the persisted product drafts, waiting to be resolved, to fetch.
+   * @param keys the keys of the persisted drafts, waiting to be resolved, to fetch.
+   * @param containerKey the key of the container, which contains the persisted draft
+   * @param clazz Class of the object contained by custom object
    * @return {@link CompletionStage}&lt;{@link Set}&gt; in which the result of its completion
    *     contains a {@link Set} that contains the matching drafts if any exist, otherwise empty.
    */
   @Nonnull
-  CompletionStage<Set<WaitingToBeResolved>> fetch(@Nonnull final Set<String> keys);
+  CompletionStage<Set<WaitingToBeResolved>> fetch(
+      @Nonnull final Set<String> keys,
+      @Nonnull final String containerKey,
+      @Nonnull Class<? extends WaitingToBeResolved> clazz);
 
   /**
    * Persists a product draft that is not ready to be resolved yet.
```

Comment:
I think if the interface is changed, the javadocs should too 😉

Reply:
daa64576f